### PR TITLE
Fixes wayfinder dispenser sprite/lowers amount of pinpointers from it/adds pinpointer recycling

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

Reduces the availability of wayfinding pinpointers through the dispenser.

Also fixed the sprite (had a hole) and used the AI status display GERTY sprites to overlay the blank screen. The dispenser has a personality now! It talks and emotes. I guarantee at least a 1% increase in customer interaction as a result.

### Other changes:
* Pinpointers now cost 65 cr
* Pinpointers can be refunded for 40 cr (to encourage recycling) or dumb costumes if it's yours and from the quirk or the dispenser is out of money
* Increases the dispenser pinpointer spawn cooldown from 2 minutes to 5
* Dispenser won't give you a pinpointer if you have one on you and the spawn cooldown will reset
* If there's a pinpointer adjacent to you the machine will point to it and the spawn cooldown will reset
* If you're on cooldown and there's a pinpointer in your view the machine will point to it and suggest you get it instead
* The pinpointers are no longer indestructible
* Pinpointers from the quirk also register their owners so if dropped I can add your name to a list

## Why It's Good For The Game

Less exploitable.

## Changelog
:cl: cacogen
tweak: The wayfinding pinpointer dispenser is different who cares. Recycle discarded pinpointers at the dispenser for 40 cr. 
/:cl:

### CODE REVIEW

**Comment by trollbreeder** [comment]
Please no more forcing the pinpointer quirk onto new people!

**Comment by Tlaltecuhtli** [comment]
the waypointers spawn in everyone's bag regardless of having or not the quirk

**Comment by cacogen** [comment]
Thought it was only assistants. It should be checking for less than 20 hours living experience and only giving them to those people. I'll get rid of it.

File: code/controllers/subsystem/processing/quirks.dm [hunk 1]
```diff
@@ -41,7 +41,3 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
 			badquirk = TRUE
 	if(badquirk)
 		cli.prefs.save_character()
-
-	// Assign wayfinding pinpointer granting quirk if they're new
-	if(cli.calc_exp_type(EXP_TYPE_LIVING) < 1200 && !user.has_quirk(/datum/quirk/needswayfinder))
```
Comment (kriskog) [inline code/controllers/subsystem/processing/quirks.dm]: cli.calc_exp_type(EXP_TYPE_CREW)

I led you astray here, EXP_TYPE_LIVING doesnt work with this proc. Crew however, will. I have confirmed it on the live servers.

**Review by kriskog — CHANGES_REQUESTED** [review]
I'll review the rest of this tomorrow.

File: code/controllers/subsystem/processing/quirks.dm (hunk 1 above)
Comment (kriskog) [reply code/controllers/subsystem/processing/quirks.dm]: Actually replace the 1200 with the define 
20 HOURS 
as well. Magic numbers and all.

File: code/game/objects/items/wayfinding.dm [hunk 2]
```diff
@@ -1,32 +1,141 @@
...
+	var/refund_amt = 40 //how much recycling a pinpointer rewards you
+	var/expression_timer
+
+/obj/machinery/pinpointer_dispenser/Initialize(mapload)
+	..()
+	var/datum/bank_account/civ_acc = SSeconomy.get_dep_account(ACCOUNT_CIV)
+	if(civ_acc)
+		synth_acc.transfer_money(civ_acc, start_bal) //float has to come from somewhere, right?
+
+	synth_acc.account_holder = src.name
```
Comment (kriskog) [inline code/game/objects/items/wayfinding.dm]: src is implied here, just name.

File: code/game/objects/items/wayfinding.dm [hunk 3]
```diff
@@ -1,32 +1,141 @@
...
+	for(var/obj/item/pinpointer/wayfinding/WP in user.GetAllContents())
+		set_expression("unsure", 20)
+		say("<span class='robot'>I know you already have a pinpointer, [user.first_name()].</span>")
+		user_spawn_cooldowns[user.real_name] = world.time + spawn_cooldown //spawn timer resets for trickers
+		return
+
+	var/msg1 = ""
+	var/msg2 = ""
+	var/msg3 = ""
+	var/msg4 = ""
```
Comment (kriskog) [inline code/game/objects/items/wayfinding.dm]: This is pretty messy. msg1 is never assigned, 2 and 3 there is no reason you cant do in a single var. 
Preferably, do all of this in a single var and just append to it as needed. So replace check thing -> add to msg2/3 var with check thing, add to msg var and do it all in the right order.

File: code/game/objects/items/wayfinding.dm [hunk 4]
```diff
@@ -1,32 +1,141 @@
...
-	pinpointers[user.real_name] = world.time + spawn_cooldown
+/datum/bank_account/wayfinding
+	add_to_accounts = FALSE
 
-/obj/item/pinpointer/wayfinding //For new players or new stations to help players find their way around
+/obj/item/pinpointer/wayfinding //Help players new to a station find their way around
 	name = "wayfinding pinpointer"
 	desc = "A handheld tracking device that points to useful places."
 	icon_state = "pinpointer_crew"
+	resistance_flags = null
```
Comment (kriskog) [inline code/game/objects/items/wayfinding.dm]: resistance_flags = NONE

**Review by kriskog — CHANGES_REQUESTED** [review]

**Comment by cacogen** [comment]
@kriskog mind reviewing this when you get the chance?

File: code/game/objects/items/wayfinding.dm [hunk 5]
```diff
@@ -1,32 +1,138 @@
...
-	if(world.time < pinpointers[user.real_name])
-		var/secsleft = (pinpointers[user.real_name] - world.time) / 10
-		to_chat(user, "<span class='warning'>You need to wait [secsleft/60 > 1 ? "[round(secsleft/60)] minute\s" : "[round(secsleft)] second\s"].</span>")
+	if(world.time < user_interact_cooldowns[user.real_name])
+		to_chat(user, "<span class='warning'>It doesn't respond.</span>")
+		return
+
+	user_interact_cooldowns[user.real_name] = world.time + interact_cooldown
+
+	for(var/obj/item/pinpointer/wayfinding/WP in user.GetAllContents())
```
Comment (kriskog) [inline code/game/objects/items/wayfinding.dm]: if(var/obj/item/pinpointer/wayfinding/WP in user.GetAllContents())
You wont be running this more than once

File: code/game/objects/items/wayfinding.dm [hunk 6]
```diff
@@ -1,32 +1,138 @@
...
+		user_spawn_cooldowns[user.real_name] = world.time + spawn_cooldown
+		user.put_in_hands(P)
+		P.owner = user.real_name
+
+/obj/machinery/pinpointer_dispenser/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/pinpointer/wayfinding))
+		var/obj/item/pinpointer/wayfinding/WP = I
+		to_chat(user, "<span class='notice'>You put \the [WP] in the return slot.</span>")
+		var/rfnd_amt
+		if(synth_acc.has_money(TRUE) && !WP.roundstart || WP.owner != user.real_name)
```
Comment (kriskog) [inline code/game/objects/items/wayfinding.dm]: Always check the cheapest one first, in this case the var. Not super important for such a rare thing, but a good habit.

if(!WP.roundstart && synth_acc.has_money(TRUE) || WP.owner != user.real_name)

File: code/game/objects/items/wayfinding.dm [hunk 7]
```diff
@@ -1,32 +1,138 @@
...
+			synth_acc._adjust_money(-rfnd_amt)
+			var/obj/item/holochip/HC = new /obj/item/holochip(user.loc)
+			HC.credits = rfnd_amt
+			HC.name = "[HC.credits] credit holochip"
+			if(istype(user, /mob/living/carbon/human))
+				var/mob/living/carbon/human/H = user
+				if(!H.put_in_hands(HC))
+					H.pulling = HC
+			else
+				user.pulling = HC
```
Comment (kriskog) [inline code/game/objects/items/wayfinding.dm]: Just drop it, dont set pulling. If they want to pull it, they can do so themselves.

**Review by kriskog — CHANGES_REQUESTED** [review]

**Comment by cacogen** [comment]
re: travis
Baby, you've been going so crazy
Lately, nothing seems to be going right

File: code/game/objects/items/wayfinding.dm (hunk 5 above)
Comment (kriskog) [reply code/game/objects/items/wayfinding.dm]: if(var/obj/item/pinpointer/wayfinding in user.GetAllContents())

**Review by kriskog — CHANGES_REQUESTED** [review]

File: code/game/objects/items/wayfinding.dm (hunk 5 above)
Comment (cacogen) [reply code/game/objects/items/wayfinding.dm]: Doesn't work, unfortunately. Neither does plain /obj/item/pinpointer/wayfinding.  But as you say, it'll only run until it finds a pinpointer. The for loop will terminate on the return, won't it?

File: code/controllers/subsystem/processing/quirks.dm [hunk 8]
```diff
@@ -2,6 +2,7 @@
 // - Quirk strings are used for faster checking in code
 // - Quirk datums are stored and hold different effects, as well as being a vector for applying trait string
 PROCESSING_SUBSYSTEM_DEF(quirks)
+#define EXP_ASSIGN_WAYFINDER 1200
```
Comment (kriskog) [inline code/controllers/subsystem/processing/quirks.dm]: This is inside something else. Top of the file

**Review by kriskog — CHANGES_REQUESTED** [review]

**Pull request merged**